testing/python/raises.py

```diff
@@ -278,3 +278,7 @@ def __class__(self):
                 with pytest.raises(CrappyClass()):
                     pass
             assert "via __class__" in excinfo.value.args[0]
+
+    def test_u(self):
```

Comment:
```
______________________________ TestRaises.test_u ______________________________

self = <raises.TestRaises object at 0x0000000007B297B8>

    def test_u(self):
        with pytest.raises(AssertionError, match=u"\u2603"):
>           assert False, u"\u2603"
E           UnicodeEncodeError: 'ascii' codec can't encode character u'\u2603' in position 0: ordinal not in range(128)
```

seems to be failing on windows CI

Reply:
I've applied this patch:

```diff
diff --git a/src/_pytest/_code/code.py b/src/_pytest/_code/code.py
index e621f3ee0..407155373 100644
--- a/src/_pytest/_code/code.py
+++ b/src/_pytest/_code/code.py
@@ -574,7 +574,7 @@ class ExceptionInfo(object):
         __tracebackhide__ = True
         value = safe_str(self.value)
         if not re.search(regexp, value):
-            assert 0, "Pattern '{!s}' not found in '{!s}'".format(regexp, value)
+            assert 0, u"Pattern {!r} not found in {!r}".format(regexp, value)
         return True
```

The problem is that even so the check still fails, when it should pass:

```
______________________________ test_u _______________________________

    def test_u():
        with pytest.raises(AssertionError, match=u"\u2603"):
>           assert False, u"\u2603"
E           AssertionError: Pattern u'\u2603' not found in '\xe2\x98\x83\nassert False'

.tmp\test_u.py:6: AssertionError
```

This will be true for any system which can't handle that unicode string using the default locale.

This solution works for me and the rest of the suite:

```
```

Reply:
though this probably triggers the same error I posted in the issue when using `--tb=native` since now we're raising a (n incorrect) unicode message in python 2 🤔

Reply:
Thanks, but actually it works:

```
 λ pytest .tmp\test_u.py  -q
.
1 passed in 0.02 seconds
```

```
λ pytest .tmp\test_u.py  -q --tb=native
.
1 passed in 0.02 seconds
```


 so I think we can go with that solution. 👍